staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -110,8 +111,52 @@ func parsePorts(ports []string) ([]ForwardedPort, error) {
 	return forwards, nil
 }
 
-// New creates a new PortForwarder.
+type listenAddress struct {
+	address     string
+	protocol    string
+	failureMode string
+}
+
+func parseAddresses(addressesToParse []string) ([]listenAddress, error) {
+	var addresses []listenAddress
+	parsed := make(map[string]listenAddress)
+	for _, address := range addressesToParse {
+		if address == "localhost" {
```

Comment:
This looks wrong. I think you need to resolve the name rather than hard coding this stuff.

Reply:
Likely you should attempt resolving any name that isn't an IP address.